### PR TITLE
feat: Rarity endpoints return prices field with prices in various currencies

### LINKED ISSUE #490: /rarities should return converted rarity prices
The endpoint only returns the prices defined in the graph. This was enough when prices were fixed in MANA, but with prices now in USD and paid in MANA, a conversion is required to be done by the endpoint.

### PR DESCRIPTION
**Pull request by fzavalia**
Closes #490

### CODE REVIEW

**Comment by fzavalia** [comment]
Requires https://github.com/decentraland/decentraland-transactions/pull/47

File: src/Rarity/Rarity.router.ts [hunk 1]
```diff
@@ -1,18 +1,97 @@
...
+        STATUS_CODES.notFound
+      )
+    }
+
+    return {
+      id: rarity.name,
+      name: rarity.name,
+      price: rarity.price.toString(),
+      maxSupply: rarity.maxSupply.toString(),
+    }
```
Comment (LautaroPetaccio) [inline src/Rarity/Rarity.router.ts:95]: What if we return the currency of the price. If `inUSD` is not set, it's not clear which is the currency it's returning the price.

File: src/Rarity/Rarity.router.ts [hunk 2]
```diff
@@ -1,18 +1,97 @@
...
 
-  async getRarities() {
+  getRarities = async (req: Request) => {
+    // If the server is still using the old rarities contract.
+    // Return rarities as they have been always returned
+    if (!isUsingRaritiesWithOracle()) {
+      return collectionAPI.fetchRarities()
+    }
+
+    const inUSD = req.query.inUSD === 'true'
```
Comment (LautaroPetaccio) [inline src/Rarity/Rarity.router.ts:27]: What if we name this query parameter `currency`? We can later have different conversions if we want to.
```suggestion
    const inUSD = req.query.currency === 'usd'
```

File: src/Rarity/Rarity.router.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    beforeEach(() => {
+      mockIsUsingRaritiesWithOracle.mockReturnValueOnce(false)
+    })
+
+    it('should fail with an endpoint not found', async () => {
+      await server.get(buildURL('/rarities/common')).expect(404)
+    })
+  })
+
+  // TODO: Tests for when rarities feature flag is enabled
```
Comment (LautaroPetaccio) [inline src/Rarity/Rarity.router.spec.ts:164]: Tests are missing here :)

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/Rarity/Rarity.router.ts (hunk 2 above)
Comment (fzavalia) [reply src/Rarity/Rarity.router.ts:27]: I like that!

**Comment by fzavalia** [comment]
@LautaroPetaccio Thanks for the feedback!
- Added missing tests.
- Change query param to currency (defaults to MANA).
- Returned rarities also include a currency field indicating in what currency the price is in.

File: src/Rarity/Rarity.router.ts [hunk 4]
```diff
@@ -1,18 +1,101 @@
...
+    return graphRarities.map((graphRarity) => {
+      // Not handling if the blockchain rarity is not present because that is checked when calling the
+      // this.getRarityFromBlockchain function.
+      const blockchainRarity = blockchainRaritiesMap.get(graphRarity.name)!
+
+      return {
+        ...graphRarity,
+        price: blockchainRarity.price,
+        originalPrice: graphRarity.price,
+        originalCurrency: Currency.USD,
```
Comment (LautaroPetaccio) [inline src/Rarity/Rarity.router.ts:49]: As "Original" as name might be a bit confusing, what if we change this to a map?

As a map:
```
price: {
'MANA': '4000000000000000000',
'USD': '10000000000000000000'
}
```

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

**Comment by fzavalia** [comment]
@LautaroPetaccio Replaced "original" fields with "prices" field containing a Record of Currency -> Price

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**